docs/developer-guide/zpool.md

```diff
@@ -0,0 +1,114 @@
+[root@headnode ~]# 
+```
+
+If a pool was not created with -B, it can still be bootable IF AND ONLY IF
+the server is bootable with BIOS. One created with -B can be bootable by
+both BIOS and EFI servers.
+
+Not all pools can be bootable.  We suggest a simple disk set for a
+ZFS-bootable Head Node.  Such sets include:
+
+- Single disk.
+- Two (mirrored) or more (raidz) same-sized disk.
+- Two or more same-sized disk with a single SSD for a log device.
+
+NOTE:  After installation a bootable pool can have log devices or log device
+mirrors added.
+
+## Installing a ZFS bootable Head Node
+
+In addition to [regular network attachment and
+configuration](https://docs.joyent.com/private-cloud/install/network-layout),
+if you wish to start booting a Head Node from a ZFS pool you should attach
+set of server disks per the introduction.  You should then use either an ISO
+installation, or if the server is iPXE installable (e.g. a bare-metal cloud
+provider), you should point at the HTTP/HTTPS-served contents of the iPXE
+install tarball.
+
+Future enhancements to the installer will allow for more flexibility in
+installing for ZFS bootable Head Nodes.  For now, the set of disks on the
+Head Node should be a simple set.
+
+### For ISO Installations
+
+The disks should be clean or otherwise unbootable.  The optical drive should
+NOT be the first tried boot device (this is ESPECIALLY true for virtual
+machines).  Also, the server (or virtual machine) should have at least two
+VLANs or actual LANs associated with its ethernet port(s).  If you are
+installing a disk-bootable CoaL, you can either follow the CoaL directions to
+setup external and admin network, OR you may configure your own prior to
+booting the ISO installer.
+
+Make sure the Triton installation ISO is in the optical drive (virtual or
+otherwise), and boot it.  Proceed to follow the installation prompts per
+[Installing Triton Elastic Container
+Infrastructure](https://docs.joyent.com/private-cloud/install).
+
+### For iPXE Installations
+
+As with ISO installations, it is important that the boot order for the server
+be the disks first, but the disks should not be bootable for the initial iPXE
+installation.
+
+An iPXE installation requires that iPXE web directory that allows a boot into
+the Triton installer also be reachable by the configured-at-installation time
+"external" network.  The iPXE web directory contains a copy of the larger ISO
+contents, which cannot fit into an iPXE boot-archive/initrd.
+
+
+## Converting from a USB-key (or from one bootable pool to another)
+
+The [piadm(1M)
+command](https://github.com/joyent/smartos-live/blob/master/man/usr/share/man/man1m/piadm.1m.md)
+can be employed to transfer the current USB-key to a bootable ZFS pool.  It
+can also be used to transfer boot contents from one pool to another.  If a
+pool is not bootable, piadm(1M) will fail.
+
+Select a POOL, and then issuing `piadm bootable -e POOL` will copy the USB
```

Comment:
Maybe 'Select a POOL, then issue `piadm bootable -e POOL` to copy the USB key contents... ... pool to be bootable. If the pool is not bootable, the `piadm` command will fail.'

Reply:
Yeah, thank you.